eth/stagedsync/bor_heimdall_shared.go

```diff
@@ -117,6 +116,207 @@ func fetchAndWriteHeimdallSpan(
 	return spanID, nil
 }
 
+func fetchRequiredHeimdallCheckpointsIfNeeded(
```

Comment:
revise the function name:
* not clear what is meant by "required"?
* there's a function fetchAndWrite, so this one is not expected to write, or it should it be fetchAndWrite as well?

alternative names: fetchAndWriteMissingHeimdallCheckpoints,  fetchAndWriteHeimdallCheckpointsIfNeeded

Reply:
I have just followed the existing language used in this file.  I guess I can change all of them to make them more consistent.